subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/DefaultXcodeProject.java

```diff
@@ -30,21 +31,17 @@
     public static final String BUILD_RELEASE = "Release";
     public static final String TEST_DEBUG = "__GradleTestRunner_Debug";
 
-    private final ConfigurableFileCollection sources;
+    private final Groups groups;
     private List<XcodeTarget> targets = new ArrayList<XcodeTarget>();
     private File locationDir;
 
     @Inject
-    public DefaultXcodeProject(FileOperations fileOperations) {
-        this.sources = fileOperations.files();
+    public DefaultXcodeProject(ObjectFactory objectFactory) {
+        this.groups = objectFactory.newInstance(Groups.class);
     }
 
-    public void addSourceFile(File sourceFile) {
-        sources.from(sourceFile);
-    }
-
-    public ConfigurableFileCollection getSources() {
-        return sources;
+    public Groups getGroups() {
```

Comment:
Is this extra layer necessary?  Couldn't we get by with just getters for sources/headers/tests/root directly on XcodeProject?

Reply:
Probably. My thinking was its private API and everything else in this class is a bit unrelated so I wanted to group everything a bit more.